includes/wc-payment-api/class-wc-payments-api-client.php

```diff
@@ -1597,8 +1597,7 @@ protected function request( $params, $api, $method, $is_site_specific = true, $u
 			$url          .= '?' . http_build_query( $params );
 			$redacted_url .= '?' . http_build_query( $redacted_params );
 		} else {
-			$headers['Idempotency-Key'] = $this->uuid();
```

Comment:
Why have we removed sending this in headers?

Reply:
It was hard to test it because this parameter is unique and mocking it is extremely hard. The point is that in unit tests, the `containts()` is used, but the same function was removed from PHPUnit 8 (or 9) which makes pipelines fails on PHP 8.* versions. To overcome this issue, the header is removed inside HTTP Client library

Reply:
If I recall correctly, the reason we added the Idempotency Key to GET, DELETE requests was to display in logs and potentially help when debugging. If we move this into the `remote_request` method, I believe we lose this benefit, making the key sort of pointless.

> The point is that in unit tests, the containts() is used, but the same function was removed from PHPUnit 8 (or 9) which makes pipelines fails on PHP 8.* versions.

I haven't looked too far into this, but couldn't we use [the `callback` constraint](https://phpunit.readthedocs.io/en/9.5/test-doubles.html#test-doubles-mock-objects-examples-subjecttest3-php) here to deal with the randomness of the Idempotency-Key header?

Reply:
> If I recall correctly, the reason we added the Idempotency Key to GET, DELETE requests was to display in logs and potentially help when debugging. If we move this into the remote_request method, I believe we lose this benefit, making the key sort of pointless.

This makes sense. However, since this is critical, I think we can create another issue to add up tests for it to avoid other devs to refactor it without knowing this knowledge. 

> I haven't looked too far into this, but couldn't we use the callback constraint here to deal with the randomness of the Idempotency-Key header?

Yeah, this seems a reasonable solution.

Reply:
@chickenn00dle I manage to fix the issue. Could you please verify it?